gitapp/lib/main.dart

```diff
@@ -46,19 +42,82 @@ class MyHomePage extends StatefulWidget {
   _MyHomePageState createState() => _MyHomePageState();
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+class _MyHomePageState extends State<MyHomePage> with TickerProviderStateMixin {
+  //FGA-3
+  final List<UserCard> _card = <UserCard>[];
+  final UsersList users = new UsersList();
+  String name = "";
+  final TextEditingController _textController = new TextEditingController();
+  bool searching =  false;
+
+  Future _getUsers(String text) async {
 
-  void _incrementCounter() {
+    _card.clear();
     setState(() {
-      // This call to setState tells the Flutter framework that something has
-      // changed in this State, which causes it to rerun the build method below
-      // so that the display can reflect the updated values. If we changed
-      // _counter without calling setState(), then the build method would not be
-      // called again, and so nothing would appear to happen.
-      _counter++;
+      searching = true;
     });
+   if(text!="")
+     {
+       name= text;
+       _textController.clear();
+       users.GetSearchedUsers(text);
+       if(users.searchedlist !=null)
+       {
+
+         for (User user in users.searchedlist)
+         {
+           UserCard card = new UserCard(
+             user:user.username,
+             image: user.avatarUrl,
+
+
+             animationController: new AnimationController(
+               duration: new Duration(milliseconds: 700),
```

Comment:
If we are using constants then please also move those milliseconds to contants

Reply:
solved